### ISSUE TITLE
Mysql2ChefGem provider conflicts with mysql2_chef_gem dependency of database cookbook

### ISSUE THREAD

**Issue by ozzyogkush** [phlipper/chef-percona#313]
When depending the `database` cookbook, I get the following error:

```
Compiling Cookbooks...
       
       ================================================================================
       Recipe Compile Error in /tmp/kitchen/cache/cookbooks/percona/libraries/provider_mysql2_chef_gem.rb
       ================================================================================
       
       TypeError
       ---------
       superclass mismatch for class Mysql2ChefGem
       
       Cookbook Trace:
       ---------------
         /tmp/kitchen/cache/cookbooks/percona/libraries/provider_mysql2_chef_gem.rb:4:in `<class:Provider>'
         /tmp/kitchen/cache/cookbooks/percona/libraries/provider_mysql2_chef_gem.rb:2:in `<class:Chef>'
         /tmp/kitchen/cache/cookbooks/percona/libraries/provider_mysql2_chef_gem.rb:1:in `<top (required)>'
       
       Relevant File Content:
       ----------------------
       /tmp/kitchen/cache/cookbooks/percona/libraries/provider_mysql2_chef_gem.rb:
       
         1:  class Chef
         2:    class Provider
         3:      # Public: Provider to install `mysql2` gem for Chef
         4>>     class Mysql2ChefGem < Chef::Provider::LWRPBase
         5:        use_inline_resources if defined?(use_inline_resources)
         6:  
         7:        def whyrun_supported?
         8:          true
         9:        end
        10:  
        11:        def action_install
        12:          converge_by "install mysql2 chef_gem and dependencies" do
        13:            recipe_eval do
```

In addition, if I omit the dependency of `mysql2_chef_gem` and attempt to use the `mysql2_chef_gem` resource provided by this cookbook, I get the following error:

```
 Recipe Compile Error in /tmp/kitchen/cache/cookbooks/pet360-magento/recipes/db_install_configure.rb
       ================================================================================
       
       NoMethodError
       -------------
       No resource or method named `mysql2_chef_gem' for `Chef::Recipe "db_import_configure"'
       
       Cookbook Trace:
       ---------------
         /tmp/kitchen/cache/cookbooks/pet360-magento/recipes/db_import_configure.rb:18:in `from_file'
         /tmp/kitchen/cache/cookbooks/pet360-magento/recipes/db_install_configure.rb:49:in `from_file'
       
       Relevant File Content:
       ----------------------
       /tmp/kitchen/cache/cookbooks/pet360-magento/recipes/db_import_configure.rb:
       
        11:    end
        12:  
        13:    # ...and bounce so nothing else in this recipe runs.
        14:    return
        15:  end
        16:  
        17:  # Install the mysql2 ruby gem
        18>> mysql2_chef_gem 'default' do
        19:    provider Chef::Provider::Mysql2ChefGem::Percona
        20:    action :install
        21:  end
        22:  
        23:  ##### BEGIN IMPORT OF DATA
        24:  # Ensure we have the /opt/sql/snapshot_storage directory so we can move the latest snapshot there.
        25:  snapshot_storage_dir = node['pet360-magento']['mysql']['snapshot_storage_dir']
        26:  directory snapshot_storage_dir do
        27:    action :create
```

**Comment by gravesb** [comment]
I believe this was fixed in #297 but it hasn't been tagged/released since that was merged in.

**Comment by ozzyogkush** [comment]
Good call!

**Comment by ozzyogkush** [comment]
I was able to get some stuff working with this commit. Any idea when the next release will be built (with this included)?

**Comment by ozzyogkush** [comment]
One thing I noticed though, I get this message when calling

```
mysql_chef_gem 'default'do
    provider Chef::Provider::MysqlChefGem::Percona
end
```

```
[2015-06-17T20:48:46+00:00] WARN: chef_gem[mysql] chef_gem compile_time installation is deprecated
[2015-06-17T20:48:46+00:00] WARN: chef_gem[mysql] Please set `compile_time false` on the resource to use the new behavior.
[2015-06-17T20:48:46+00:00] WARN: chef_gem[mysql] or set `compile_time true` on the resource if compile_time behavior is required.
```

This class should allow you to pass along any attributes the `chef_gem` resource might want, such as `compile_time`.

**Comment by ozzyogkush** [comment]
Noticed that this was actually released on June 4th in v0.16.1. Closing. Thanks!
